nautobot/ipam/tests/test_filters.py

```diff
@@ -1249,3 +1254,278 @@ def test_virtual_machine(self):
+        self.assertEqual(IPAddress.objects.filter(host__iendswith="0.2").count(), 1)
+        self.assertEqual(IPAddress.objects.filter(host__iendswith="0.1").count(), 2)
+        self.assertEqual(IPAddress.objects.filter(host__iendswith="0.50").count(), 0)
+
+        self.assertEqual(IPAddress.objects.filter(host__endswith="8::1").count(), 1)
+        self.assertEqual(IPAddress.objects.filter(host__endswith="8::5").count(), 0)
+        self.assertEqual(IPAddress.objects.filter(host__iendswith="8::1").count(), 1)
+        self.assertEqual(IPAddress.objects.filter(host__iendswith="8::5").count(), 0)
+
+    @skipIf(
+        connection.vendor == "postgresql",
+        "Not currently supported on postgresql",
+    )
+    def test_startswith(self):
+        self.assertEqual(IPAddress.objects.filter(host__startswith="10.0.0.").count(), 5)
+        self.assertEqual(IPAddress.objects.filter(host__startswith="10.0.0.1").count(), 2)
+        self.assertEqual(IPAddress.objects.filter(host__startswith="10.50.0.").count(), 0)
+        self.assertEqual(IPAddress.objects.filter(host__istartswith="10.0.0.").count(), 5)
+        self.assertEqual(IPAddress.objects.filter(host__istartswith="10.0.0.1").count(), 2)
+        self.assertEqual(IPAddress.objects.filter(host__istartswith="10.50.0.").count(), 0)
+
+        self.assertEqual(IPAddress.objects.filter(host__startswith="2001:db8::").count(), 3)
+        self.assertEqual(IPAddress.objects.filter(host__startswith="2001:db8::1").count(), 1)
+        self.assertEqual(IPAddress.objects.filter(host__startswith="2001:db8::5").count(), 0)
+        self.assertEqual(IPAddress.objects.filter(host__istartswith="2001:db8::").count(), 3)
+        self.assertEqual(IPAddress.objects.filter(host__istartswith="2001:db8::1").count(), 1)
+        self.assertEqual(IPAddress.objects.filter(host__istartswith="2001:db8::5").count(), 0)
+
+    @skipIf(
+        connection.vendor == "postgresql",
+        "Not currently supported on postgresql",
+    )
+    def test_regex(self):
+        self.assertEqual(IPAddress.objects.filter(host__regex=r"10\.(.*)\.1").count(), 2)
+        self.assertEqual(IPAddress.objects.filter(host__regex=r"10\.(.*)\.4").count(), 1)
+        self.assertEqual(IPAddress.objects.filter(host__regex=r"10\.(.*)\.50").count(), 0)
+        self.assertEqual(IPAddress.objects.filter(host__iregex=r"10\.(.*)\.1").count(), 2)
+        self.assertEqual(IPAddress.objects.filter(host__iregex=r"10\.(.*)\.4").count(), 1)
+        self.assertEqual(IPAddress.objects.filter(host__iregex=r"10\.(.*)\.50").count(), 0)
+
+        self.assertEqual(IPAddress.objects.filter(host__regex=r"2001(.*)1").count(), 1)
+        self.assertEqual(IPAddress.objects.filter(host__regex=r"2001(.*)5").count(), 0)
+        self.assertEqual(IPAddress.objects.filter(host__iregex=r"2001(.*)1").count(), 1)
+        self.assertEqual(IPAddress.objects.filter(host__iregex=r"2001(.*)5").count(), 0)
+
+
+class PrefixFilterTestCase(TestCase):
```

Comment:
Should we have a clone of this test case that performs the same tests, but for `Aggregate`?

Reply:
I thought about it, but didn't see the value since my only goal is to verify the custom lookup. I can do it if requested, just making sure I see the value.

Reply:
I'll take silence as agreement, feel free to unresolve :)